tests/foreman/ui/test_computeresource_vmware.py

```diff
@@ -538,6 +538,103 @@ def test_positive_access_vmware_with_custom_profile(self):
 
     @run_only_on('sat')
     @tier2
+    def test_positive_apply_vmware_with_custom_profile_to_host(self):
+        """Associate custom default (3-Large) compute profile with hostgroup
+        and then inherit it to the host
+
+        :id: c16c6d42-3950-46a7-bfe6-5e19bcfa29d0
+
+        :setup: vmware hostname and credentials.
+
+        :steps:
+
+            1. Create a compute resource of type vmware.
+            2. Provide valid hostname, username and password.
+            3. Select the created vmware CR.
+            4. Click Compute Profile tab.
+            5. Edit (3-Large) with valid configurations and submit.
+            6. Create new host group with custom profile
+            7. Open new host page and put host group name into corresponding
+               field
+            8. Check that compute profile is inherited and then switch to
+               Virtual Machine tab
+
+        :expectedresults: All fields values for Virtual Machine tab are
+            inherited from custom profile and have non default values
+
+        :Caseautomation: Automated
+
+        :CaseLevel: Integration
+        """
+        org = entities.Organization().create()
+        parameter_list = [
+            ['VCenter/Server', self.vmware_url, 'field'],
+            ['Username', self.vmware_username, 'field'],
+            ['Password', self.vmware_password, 'field'],
+            ['Datacenter', self.vmware_datacenter, 'special select'],
+        ]
+        name = gen_string('alpha')
+        hg_name = gen_string('alpha')
+        with Session(self) as session:
+            make_resource(
+                session,
+                name=name,
+                provider_type=FOREMAN_PROVIDERS['vmware'],
+                orgs=[org.name],
+                org_select=True,
+                parameter_list=parameter_list
+            )
+            self.compute_resource.set_profile_values(
+                name, COMPUTE_PROFILE_LARGE,
+                cpus=3,
+                corespersocket=4,
+                memory=2048,
+                cluster=VMWARE_CONSTANTS.get('cluster'),
+                folder=VMWARE_CONSTANTS.get('folder'),
+            )
+            self.assertIsNotNone(self.compute_resource.search(name))
+            make_hostgroup(
+                session,
+                name=hg_name,
+                organizations=[org.name],
+                parameters_list=[
+                    ['Host Group', 'Compute Profile', COMPUTE_PROFILE_LARGE],
+                ],
+            )
+            self.hosts.navigate_to_entity()
+            self.hosts.click(locators['host.new'])
+            self.hosts.assign_value(locators['host.organization'], org.name)
+            self.hosts.assign_value(locators['host.host_group'], hg_name)
+            self.hosts.click(locators['host.deploy_on'])
```

Comment:
BZ  say: 1249744
 i. Select Deploy on VMware compute resource
 ii. Select Host Group 

Additional info:
This issue is not reproduced if the Host Group is selected first and then the Deploy on VMware compute profile

Reply:
You can't do that physically, or at least can't do that anymore (which is good as it is logically correct), because selecting host group will automatically resets most of the fields to inherited values or just to default values
Of course, I can do:
i. Select Deploy on VMware compute resource
ii. Select Host Group
iii.  Select Deploy on VMware compute resource
but it doesn't make any sense

Reply:
@oshtaier can you put a comment about that at that position (just for history of this test and it's connection with the BZ)